First changes.

michalkrzywicki opened the pull request.

michalkrzywicki committed:
First changes.

Vakrim commented on src/App.tsx:
```diff
@@ -1,25 +1,23 @@
-import './App.css';
+import { TodoListItem } from "./TodoListItem";
+
+const todos: Todo[] = [
```
great start 🎉 👍 

Next steps would be moving those todos to state (`useState`) and adding input for creating new todos :)

michalkrzywicki replied:
UseState now in action, sir. Input for new actions will be available soon. Please be patient.

Vakrim commented on src/App.tsx:
```diff
@@ -1,25 +1,23 @@
-    </div>
+  return(
+      <ul>
+        <TodoListItem todo={todos[0]} />
+        <TodoListItem todo={todos[1]} />
```
https://beta.reactjs.org/learn/rendering-lists

Vakrim commented on src/types.d.ts:
```diff
@@ -0,0 +1,4 @@
+interface Todo {
```
`.d.ts` files are for external libraries. 

In most cases this should be `types.ts` and this interface should be exported

```ts
export interface Todo {
```

and import in other files
```ts
import { Todo } from './types'
```

Vakrim commented on src/index.tsx:
```diff
@@ -1,19 +1,10 @@
 
-const root = ReactDOM.createRoot(
-  document.getElementById('root') as HTMLElement
+ReactDOM.render(
```
any reason why you replaced this code?

`ReactDOM.render` is deprected AFAIK

michalkrzywicki committed:
First fixes to broken things.

1. Rename file for types.
2. Use map instead of manually send things.
3. Import types to every file it needed.

Vakrim commented on src/App.tsx:
```diff
@@ -1,25 +1,24 @@
-    </div>
+  return(
+      <ul>
+          {todos.map((data) => <TodoListItem todo={data} />)}
```
keys are quite important part of making lists. You may not see the difference now, but it may bite you near feature :P https://beta.reactjs.org/learn/rendering-lists#keeping-list-items-in-order-with-key

michalkrzywicki committed:
UseState in action.

michalkrzywicki committed:
Add new component for rendering list.

michalkrzywicki committed:
UseState and form for adding new item.
